src/Lean/Message.lean

```diff
@@ -166,6 +183,11 @@ partial def formatAux : NamingContext → Option MessageDataContext → MessageD
     msg := f!"{msg} {(← formatAux nCtx ctx header).nest 2}"
     let children ← children.mapM (formatAux nCtx ctx)
     return .nest 2 (.joinSep (msg::children.toList) "\n")
+  | nCtx, ctx?,      ofLazy pp _             => do
+    let dyn ← pp (ctx?.map (mkPPContext nCtx))
+    let some msg := dyn.get? MessageData
+      | throw <| IO.userError "MessageData.ofLazy: expected MessageData in Dynamic"
```

Comment:
I would expect this to panic. Why is it an IO exception?

Reply:
It might also be nice, in case this ever does happen, to include something like `expected MessageData in Dynamic but got '{dyn.typeName}'`, but we can always put that in if it's ever a problem